tests/manage/z_cluster/cluster_expansion/test_delete_pod.py

```diff
@@ -0,0 +1,124 @@
+                    logging.info("New osd pods in status running")
+                    break
+        except TimeoutExpiredError:
+            logging.warning(
+                f"New osd pods are not in status running after {timeout} seconds"
+            )
+
+    @pytest.mark.parametrize(
+        argnames="workload_storageutilization_rbd, resource_name, resource_id, is_kill_resource_repeatedly",
+        argvalues=[
+            pytest.param(
+                *[(0.11, True, 120), constants.ROOK_OPERATOR, 0, False],
+                marks=pytest.mark.polarion_id("OCS-1206")
+            ),
+            pytest.param(
+                *[(0.11, True, 120), constants.MON_DAEMON, 0, True],
+                marks=pytest.mark.polarion_id("OCS-1207")
+            ),
+        ],
+        indirect=["workload_storageutilization_rbd"],
+    )
+    def test_add_capacity_with_resource_delete(self, workload_storageutilization_rbd, resource_name,
+                                               resource_id, is_kill_resource_repeatedly):
+        """
+        The function get the resource name, and id.
+        The function adds capacity to the cluster, and then delete the resource while
+        storage capacity is getting increased.
+        Args:
+            resource_name (str): the name of the resource to delete
+            resource_id (int): the id of the resource to delete
+            is_kill_resource_repeatedly (bool): If True then kill the resource repeatedly. Else, if False
+                delete the resource only once.
+        """
+        used_percentage = get_percent_used_capacity()
+        logging.info(f"storageutilization is completed. used capacity = {used_percentage}")
+
+        max_osds = 15
+        osd_pods_before = pod_helpers.get_osd_pods()
+        number_of_osd_pods_before = len(osd_pods_before)
+        if number_of_osd_pods_before >= max_osds:
+            pytest.skip("We have maximum of osd's in the cluster")
+
+        d = Disruptions()
+        d.set_resource(resource_name)
+
+        self.new_pods_in_status_running = False
+
+        osd_size = storage_cluster.get_osd_size()
+        logging.info(f"Adding one new set of OSDs. osd size = {osd_size}")
+        storagedeviceset_count = storage_cluster.add_capacity(osd_size)
+        logging.info("Adding one new set of OSDs was issued without problems")
+
+        # Wait for new osd's to come up. After the first new osd in status Init - delete the resource.
+        # After deleting the resource we expect that all the new osd's will be in status running,
+        # and the delete resource will be also in status running.
+        pod_helpers.wait_for_new_osd_pods_to_come_up(number_of_osd_pods_before)
+        logging.info(f"Delete a {resource_name} pod while storage capacity is getting increased")
+        if is_kill_resource_repeatedly:
+            logging.info("kill resource repeatedly chosen")
+            with ThreadPoolExecutor() as executor:
```

Comment:
Nice!!

BTW, is ThreadPoolExecutor needed here if you are doing submit only once?

Reply:
I am not sure. Do you have a better idea?